lib/vrl/stdlib/src/get_host_ip.rs

```diff
@@ -0,0 +1,138 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct GetHostIp;
+
+impl Function for GetHostIp {
+    fn identifier(&self) -> &'static str {
+        "get_host_ip"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "interface",
+                kind: kind::BYTES,
+                required: false,
+            },
+            Parameter {
+                keyword: "family",
+                kind: kind::BYTES,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let families = vec![value!("IPv4"), value!("IPv6")];
+        let interface = arguments.optional("interface");
+        let family = arguments
+            .optional_enum("family", &families)?
+            .map(|v| v.try_bytes().expect("family not bytes"));
+        Ok(Box::new(GetHostIpFn { interface, family }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "valid",
+                source: r#"get_host_ip!() != null"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "IPv4",
+                source: r#"get_host_ip!(family: "IPv4") != null"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "interface",
+                source: r#"get_host_ip!(interface: "eth0") != null"#,
```

Comment:
I imagine this test would fail on my machine since I don't have an `eth0`. Would it work with `lo` since I imagine that's more consistently named among machines?

I could see us wanting a `get_available_interfaces` function at some point.

Reply:
Oh, I see you are skipping the running of this test, so the above won't be a problem..

Reply:
Yeah, I ended up skipping these examples in the VRL tests since they are machine dependent.

Reply:
👍 to a `get_available_interfaces` in the future though.